src/validation.cpp

```diff
@@ -2632,6 +2632,12 @@ bool Chainstate::DisconnectTip(BlockValidationState& state, DisconnectedBlockTra
 
     m_chain.SetTip(*pindexDelete->pprev);
 
+    // if we are roll-backing the chain, rollback the best header as well
+    if (m_chainman.m_best_header && m_chainman.m_best_header->GetBlockHash() == pindexDelete->GetBlockHash()) {
+        assert(pindexDelete->pprev);
+        m_chainman.m_best_header = pindexDelete->pprev;
+    }
```

Comment:
Tests still pass with this chunk removed. I wonder if checking that getblockchaininfo `headers == blocks` isn't enough? Might need to use `getchaintips` to cover.